### ISSUE TITLE
3-1-17 source build: Extension Viewlet  cannot search, install any extensions (NO Proxy)

### ISSUE THREAD

**Issue by cleidigh** [Microsoft/vscode#21814]
Development Environment:

OS: Windows 7 Professional SP1 (VM under OSX)
Python: ActiveState 2.7.5

VS Code: 1.10.1 master release
Visual Studio 15 Community \ C++ Tools \ SDK 8.1
VS Code: 1.10.1 (Code OSS Build environment)
VS Code Insiders 1.10.0 2-27 build: (Test environment)
VS Code: master clone 2-28 and 3-1  date tests

Node: 6.9.1
NPM:  3.10.8

Description Of Bug:

VCS does not search, download, install any extensions
with the extension wiget. Entering anything in the search bar
results in no extensions found.

If I manually copy an extension from the extension directory
of VC code to the extension directory of .vscode-oss these
extensions will show up , activate and function. Besides
vscode-icons I also copied ftp-sync to test a network extension
this succeeded without problems. Still no ability to search download marketplace

I have no proxy, working from home.  Looking at other posted issues
they seem similar but without the proxy involved.
Other VSC release downloads have no similar problems with multiple
Code release versions.

Inspecting the developer tools console output shows no errors.
I have not gotten debugging of vscode  yet, so I have not
been able to inspect and debug the code directly yet
(anxious to do so).

Steps to reproduce:

git clone https://github.com/microsoft/vscode
.\scripts\npm install
.\scripts\code
(successful build - launches Code OSS Dev)
View extensions
(enter any search for an extension - no results)

A successful package exhibits the same bug
.\node_modules\.bin\gulp vscode-win32

manually copied extensions appear in the widget
and function (except for an update attempt)

Cheers Christopher

**Comment by sandy081** [comment]
Browsing extension gallery is not enabled in Out of Sources build hence you cannot query for extensions in Extensions viewlet. 

@joaomoreno FYI

**Comment by cleidigh** [comment]
@sandy081 
@joaomoreno 


Thanks
I just got this working last night.  After getting incremental building and  debugging finally working correctly.  I did debugging on the extension files and found the references to the URLs omitted
from product.json which disables recommendations locally and all marketplace comms. added
URLs and everything works great now.

Not sure why this is not documented unless I missed it somewhere. VSC Dev without extensions
can't go very far ;-(

Anyway great learning experience to start contributing..

Christopher

https://github.com/Microsoft/vscode/issues/17574
